pageserver/src/virtual_file.rs

```diff
@@ -580,30 +580,67 @@ mod tests {
     use std::sync::Arc;
     use std::thread;
 
-    // Helper function to slurp contents of a file, starting at the current position,
-    // into a string
-    fn read_string<FD>(vfile: &mut FD) -> Result<String, Error>
-    where
-        FD: Read,
-    {
-        let mut buf = String::new();
-        vfile.read_to_string(&mut buf)?;
-        Ok(buf)
+    enum MaybeVirtualFile {
```

Comment:
If this was non-test code I'd probably advocate to make this a trait & impl it for both types, rather than using an enum wrapper with if statements.  I don't feel strongly in this context though.

Reply:
Previously the API we expose here was exposed via a trait, but async fn doesn't support traits yet, especially not the builtin traits. There is the async-trait crate but for that we'd have to redefine the builtin traits that expose this, and it would mean more code in total.